app/index.js

```diff
@@ -15,22 +15,44 @@ var AspnetGenerator = yeoman.generators.Base.extend({
       defaults: false,
       desc: 'Use the Grunt JavaScript task runner instead of Gulp in web projects.'
     });
+
+    this.argument('type', { type: String, required: false, desc: 'the project type to create' });
+    this.argument('applicationName', { type: String, required: false, desc: 'the name of the application' });
   },
 
 
-  init: function() {
+  init: function () {
     this.log(yosay('Welcome to the marvellous ASP.NET 5 generator!'));
     this.templatedata = {};
   },
 
-  askFor: function() {
-    var done = this.async();
+  checkProjectType: function () {
```

Comment:
The same as with `buildTemplateData`:
https://github.com/OmniSharp/generator-aspnet/pull/555/files#r51627725
It needs to be private/internal helper method